src/client/api/pythonApi.ts

```diff
@@ -201,14 +201,26 @@ const ProductMapping: { [key in Product]: JupyterProductToInstall } = {
 /* eslint-disable max-classes-per-file */
 @injectable()
 export class PythonInstaller implements IPythonInstaller {
+    private readonly _onInstalled = new EventEmitter<{ product: Product; resource?: InterpreterUri }>();
+    public get onInstalled(): Event<{ product: Product; resource?: InterpreterUri }> {
+        return this._onInstalled.event;
+    }
     constructor(@inject(IPythonApiProvider) private readonly apiProvider: IPythonApiProvider) {}
 
     public install(
         product: Product,
         resource?: InterpreterUri,
         cancel?: CancellationToken
     ): Promise<InstallerResponse> {
-        return this.apiProvider.getApi().then((api) => api.install(ProductMapping[product], resource, cancel));
+        return this.apiProvider
+            .getApi()
+            .then((api) => api.install(ProductMapping[product], resource, cancel))
+            .then((result) => {
+                if (result === InstallerResponse.Installed) {
```

Comment:
Any time a module is installed we want to fetch this once again.
Else we keep the same items in the case.
Should we blow away the cache evey `n` hours?
Feels like we should (incase user keeps this open).
But if the user installs/uninstalls from CLI, there's not much we can do.
Don't want to fech very time, that will be very slow.

Reply:
I'm not sure exactly what you mean by this comment.